src/cpu/flexcpu/simple_dataflow_thread.cc

```diff
@@ -856,17 +805,18 @@ SDCPUThread::onPCTranslated(weak_ptr<InflightInst> inst, Fault fault,
     DPRINTF(SDCPUThreadEvent, "Received PC translation (va: %#x, pa: %#x)\n",
             vaddr, req->getPaddr());
 
-    auto callback = [this, inst, vaddr](uint8_t* data) {
-        bufferInstructionData(vaddr, data);
-        attemptFetch(inst);
+    auto callback = [this, next_pc, vaddr](uint8_t* data) {
+        if (fetchPC == next_pc) {
```

Comment:
Didn't we run into issues with this if you had two outstanding requests to the same PC?

Reply:
I think a better way to manage this is to keep a flag/management object for outstanding requests to the CPU, and to kill the object and the corresponding request during any squash. Then restart advancing. This was one of the jobs the empty shared_ptr<InflightInst> served when it was in the buffer, which is still an important job to do, even if we don't like having InflightInsts early.